cleo_plugins/Text/Text.cpp

```diff
@@ -68,7 +70,15 @@ class Text
 		CLEO_RegisterCallback(eCallbackId::GameEnd, OnGameEnd);
 
 		// install hooks
-		MemPatchJump(0x006A0050, &HOOK_CTextGet); // FUNC_CText__Get from CText.cpp
+		patchCTextGet = MemPatchJump(0x006A0050, &HOOK_CTextGet); // FUNC_CText__Get from CText.cpp
+	}
+
+	~Text()
+	{
+		CLEO_UnregisterCallback(eCallbackId::GameBegin, OnGameBegin);
+		CLEO_UnregisterCallback(eCallbackId::GameEnd, OnGameEnd);
+
+		patchCTextGet.Apply(); // undo hook
```

Comment:
just FYI, there might be another hook installed on top of CLEO and reverting back to original may affect it. But since it happens on game exit, the impact should be minimal

Reply:
All those registers and patches should be handled by some CLEO plugin guard.
Then cleanup would happen automatically in destructor.